railties/lib/rails/generators/app_base.rb

```diff
@@ -422,6 +427,29 @@ def os_supports_listen_out_of_the_box?
         /darwin|linux/.match?(RbConfig::CONFIG["host_os"])
       end
 
+      def target_rails_prerelease(self_command = "new")
+        gemfile = "rails_prerelease.gemfile"
+
+        if rails_prerelease? && !ENV["BUNDLE_GEMFILE"]&.end_with?(gemfile)
+          Dir.mktmpdir do |dir|
```

Comment:
We need to require `tmpdir` in this file otherwise it will fail.

Reply:
Ack, I must have mistakenly removed that during code cleanup.  I am trying a different approach now that doesn't require a tmp dir, but if I switch back to the original approach, I'll be sure to add `require "tmpdir"`.